### ISSUE TITLE
docker-compose build fails on Apple Silicon (M1)

### ISSUE THREAD

**Issue by benkaiser** [Neutron-Creative/Singlelink#179]
Trying to clone the repo and running `docker-compose up` fails, first due to `python` not being available, and then after adding that, it's due to an older version of `glibc` that fibers/sharp depends on. Here is the relevant github issue about the sharp/glibc dependency:
https://github.com/lovell/sharp/issues/2482
Once the `node` docker releases images based on the next version of Debian ("Bullseye" / 11), we should be able to switch to that, but for now I'm having to try and work out building an Ubuntu 20.04 based image, installing node+node-gyp build-time dependencies.

**Comment by benkaiser** [comment]
This is as far as I got trying to fix it, looks like the process of nuxt-ts build is getting Killed for some reason.
https://github.com/benkaiser/Singlelink/tree/m1-not-working

**Comment by NavidK0** [comment]
Make sure you have enough RAM available during the build process if you are building the Dockerfile yourself. The process of actually building takes a lot more RAM than I'd like, but this won't get fixed for quite some time.

Another thing to note is that Singlelink v3 is right around the corner ([Beta v3 Branch](https://github.com/Neutron-Creative/Singlelink/tree/beta-v3)). The project has changed substantially from v2 to v3, so I would try that branch instead. Beta v3 will be pushed to the main branch very soon (it is basically finished, just needs some documentation written).

**Comment by benkaiser** [comment]
Hey @NavidK0 ,

I actually gave up on my M1 for the day and switched to my windows desktop (hence the other issue filed).

I tried making sure the host has enough memory (7GB free), and passing a flag in the docker-conpose to up the memory to 8GB, and it was still failing at that "Killed" line. So not quite sure what was killing it.

I'll give it another shot on v3 when I get a chance this week.

**Comment by NavidK0** [comment]
> Hey @NavidK0 ,
> 
> I actually gave up on my M1 for the day and switched to my windows desktop (hence the other issue filed).
> 
> I tried making sure the host has enough memory (7GB free), and passing a flag in the docker-conpose to up the memory to 8GB, and it was still failing at that "Killed" line. So not quite sure what was killing it.
> 
> I'll give it another shot on v3 when I get a chance this week.

Fair enough. We can switch the Dockerfile build image to Ubuntu for now to work around the Debian glibc issue.

**Comment by benkaiser** [comment]
@NavidK0 btw I just saw an update that docker node images now support debian bullseye. So switching to that should fix the build chain issues I was facing.

I'll test again at some point when I've got some time
